src/wcl/doc.h

```diff
@@ -263,7 +263,17 @@ class doc_builder {
   bool has_newline() const { return state.newline_count > 0; }
 
   doc build() && {
-    assert(docs.size() > 0);
+    if (docs.size() == 0) {
+      return doc::lit("");
+    }
+
+    if (docs.size() == 1) {
+      doc copy = std::move(docs[0]);
+      docs = {};
+      state = {};
+      return copy;
+    }
```

Comment:
These edge cases should be handled in `merge` if merge is written correctly.

Reply:
The `0` edge case is needed so that we can have "epsilon else" in the IfElseAction since we try to build an empty builder. The `1` case is probably not needed. I just added it without really thinking about it

Reply:
I didn't say it wasn't needed, I said it should be handled in merge ;) merge should be able to merge any length of array, 0, 1 or higher. Shame on me for not checking on that sooner.

Reply:
Yeah, the `1` isn't needed, but `0` is since you can't merge from `0` to `-1` and even if you could, there is nothing in the array to return

Reply:
you can define merge to return the empty doc in that case